tests/integration/optimizers/test_optimizer.py

```diff
@@ -16,56 +19,80 @@
 }
 
 
+@pytest.fixture
+def config(tmpdir):
+    root = os.path.join('tests', 'integration', 'optimizers')
+    os.environ['JINA_OPTIMIZER_FLOW_JAML_PATH'] = os.path.join(root, 'flow.yml')
+    os.environ['JINA_OPTIMIZER_DATA_PATH'] = os.path.join(root, 'data.jsonlines')
+    os.environ['JINA_OPTIMIZER_PARAMETER_PATH'] = os.path.join(root, 'parameter.yml')
+    os.environ['JINA_OPTIMIZER_WORKSPACE_DIR'] = str(tmpdir)
+    os.environ['JINA_OPTIMIZER_OUTPUT_FILE'] = os.path.join(tmpdir, 'best_parameters.yml')
+    yield
+    del os.environ['JINA_OPTIMIZER_FLOW_JAML_PATH']
+    del os.environ['JINA_OPTIMIZER_DATA_PATH']
+    del os.environ['JINA_OPTIMIZER_PARAMETER_PATH']
+    del os.environ['JINA_OPTIMIZER_WORKSPACE_DIR']
+    del os.environ['JINA_OPTIMIZER_OUTPUT_FILE']
+
+
 def document_generator(num_doc):
     for _ in range(num_doc):
         doc = Document(content='hello')
         groundtruth_doc = Document(content='hello')
         yield doc, groundtruth_doc
 
 
-def test_optimizer(tmpdir):
+@pytest.mark.parametrize('use_output_file', (True, False))
+def test_optimizer(tmpdir, use_output_file):
     eval_flow_runner = SingleFlowRunner(
-        flow_yaml='tests/integration/optimizers/flow.yml',
+        flow_yaml=os.path.join('tests', 'integration', 'optimizers', 'flow.yml'),
```

Comment:
This is done for our Windows users, right? I find it seriously ugly and waaaay less readable. Do we care in the rest of the code base about it? I thought no.

Reply:
I know that it is in general good practice to use os.path.join to be agnostic to the platform. I started a discussion on slack.

Reply:
we care almost everywhere I think, at least we used to use `os.path.join` everywhere

Reply:
I changed it back for now